### PR TITLE
(OraklNode) Refactor request package

### PR DESCRIPTION
**Pull request by nick-bisonai**
# Description

Implement functional options pattern to request package

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Deployment

- [ ] Should publish npm package
- [x] Should publish Docker image

### CODE REVIEW

File: node/pkg/utils/request/request.go [hunk 1]
```diff
@@ -12,17 +12,64 @@ import (
...
-	response, err := UrlRequestRaw(urlEndpoint, method, requestBody, headers, proxy)
+
+	config := RequestConfig{
+		Timeout: 2 * time.Second,
+		Method:  "GET",
+	}
+	for _, opt := range opts {
+		opt(&config)
+	}
+	response, err := requestRaw(config)
```
[the remark this answers is not in the document]
Comment (nick-bisonai) [reply node/pkg/utils/request/request.go:72]: yes please, please write test code for edge cases and error handling scenarios you have mentioned

File: node/pkg/utils/request/request.go [hunk 2]
```diff
@@ -52,26 +99,37 @@
...
+		Timeout: 2 * time.Second,
+		Method:  "GET",
+	}
+	for _, opt := range opts {
+		opt(&config)
+	}
+	return requestRaw(config)
+}
+
+func requestRaw(config RequestConfig) (*http.Response, error) {
```
[the remark this answers is not in the document]
Comment (nick-bisonai) [reply node/pkg/utils/request/request.go:113]: how can the proxy handling section be refactored?
